lib/core/include/scipp/core/element/histogram.h

```diff
@@ -65,14 +66,17 @@ static constexpr auto histogram = overloaded{
         const auto [offset, nbin, scale] = core::linear_edge_params(edges);
         for (scipp::index i = 0; i < scipp::size(events); ++i) {
           const auto x = events[i];
+          // Explicitly check for x outside edges here as otherwise we
+          // may run into an integer overflow when converting the "bin"
+          // computation result to `Index`.
+          if (x < edges.front() || x >= edges.back())
+            continue;
```

Comment:
Given that this fix is nearly identical in `bin` and `histogram`, is it possible to extract the common code into a function that computes the target `bin`?

Reply:
>     * 5-10% for `bin`

Scratch that, the implementation in this PR has the 5% regression (no visible difference when multi-threading is on), but extracting the function seems to perform as the original. Need to investigate more.

Reply:
I rewrote it slightly, and see no significant performance impact. There *may* be a very small slowdown, in particular for `hist` of the order of 5-10%, but I doubt it would be relevant in practice (i.e., with multi-threading).

Reply:
Looks good